src/pages/JoinPage.js

```diff
@@ -1,7 +1,110 @@
-import React from 'react'
+import React, { useState } from 'react'
+import { Flex, Button, Box, Link } from '@chakra-ui/react'
+import { Link as ReactLink, useNavigate } from 'react-router-dom'
+import FormInput from '../components/UI/FormInput'
+import { darken } from '@chakra-ui/theme-tools'
+import styles from '../styles/LoginPage.module.css'
+import FormWrapper from '../components/UI/FormWrapper'
+import { useForm } from 'react-hook-form'
+import { createProfile, uploadProfileImage } from '../api/profile'
+import { createUser } from '../api/auth'
+import { useAuthContext } from '../hooks/auth-context'
+import { createCompany } from '../api/company'
 
 const RegisterPage = () => {
-   return <div>Join Page</div>
+  const { register, handleSubmit } = useForm()
+  const [loading, setLoading] = useState()
+  const [errorRegister, setErrorRegister] = useState('')
+  const { login } = useAuthContext()
+  const navigate = useNavigate()
+  const onSubmit = async (data) => {
+    setLoading(true)
+    try {
+      const user = await createUser({
+        username: data.name,
+        email: data.email,
+        password: data.password
+      })
+
+      const companyId = await createCompany({ name: `${data.name}'s company` }, user.jwt)
+
+      const profileId = await createProfile({
+        status: 'pending',
+        name: data.name,
+        profilePhoto: null,
+        user: `${user.id}`,
+        company: companyId,
+        answers: null
+      })
+
+      await uploadProfileImage(profileId, data.profileImg[0])
```

Comment:
We should first upload the photo, and use the uploaded photo id when creating the profile :)